### ISSUE TITLE
Changing ssh port?

### ISSUE THREAD

**Issue by CharlesSchaack** [ericbarch/dockuwiki#1]
What would I need to do to get this setup to recognize a non standard ssh port? I'm getting connection refused because it's knocking on 22.

**Comment by ericbarch** [comment]
This is definitely something that should be supported, BUT in the meantime you can make it work by passing in the parameters like this:

-e SSH_DOMAIN="-p 22222 myserver.mydomain" -e REMOTE_URL=ssh://git@myserver.mydomain:22222/wiki.git

**Comment by CharlesSchaack** [comment]
still receiving the following:

```
cloning wiki...
Cloning into '/home/wiki/web'...
ssh: connect to host 192.168.1.13 port 22: Connection refused
fatal: Could not read from remote repository.
```

 unless that error is hard coded as 22? 

I have a instance of gogs running at that local IP. The command I'm running is: 

`docker run -d --restart=always --name=wiki -e SSH_DOMAIN="-p 2222 192.168.1.13" -e REMOTE_URL=git@192.168.1.13:2222/Charles/wiki.git -p 3001:3001 ericbarch/dockuwiki`

**Comment by ericbarch** [comment]
Hmm, are you starting with a fresh docker container when you do that or starting one that's already been created? I'm also using a gogs server myself and not seeing that behavior.

**Comment by CharlesSchaack** [comment]
Not too experienced with docker, I'm running "docker rm -f wiki" to delete the container and then recreating using the docker run command above.

My instance of gogs is also within a docker container. I assume you have the same setup?

**Comment by ericbarch** [comment]
Oh! Try putting the "ssh://" in front of your git@ URL. Without that, git won't know how to parse the port.

**Comment by CharlesSchaack** [comment]
Doh! 👍 

That fixed my problem. Thank you for helping me debug!

**Comment by CharlesSchaack** [comment]
Seem to be having trouble with the public key. I copy it into gogs but dockuwiki complains "permission denied (public key)". I also tried a normal clone using ssh and was denied too. I tried adding an SSH key which I use for other things to gogs and I receive the same message even when using that.

**Comment by ericbarch** [comment]
Huh. SSH keys tend to be very sensitive in regards to formatting. Did you copy exactly from the beginning of the "ssh-rsa" to the end with "wiki@xxxxxxxxxxxx" with no extra whitespace? Am I correct in assuming you added the key to your Gogs user account that owns the wiki git repo?

**Comment by CharlesSchaack** [comment]
I tried RDPing in and copying yet still get the same results. Just wanted to make sure as I have been using putty to copy the key so I thought maybe formatting was getting messed up. I am making sure to copy from beginning to end as you said without whiespaces though. I did add it to my gogs useraccount that the repository is created under.

**Comment by CharlesSchaack** [comment]
http://pastebin.com/8BHbthDF

is the key im ending up with.


http://pastebin.com/BBSb4WYC

is the whole log output from "docker logs wiki"

**Comment by CharlesSchaack** [comment]
sorry didn't mean to close.

**Comment by ericbarch** [comment]
Yeah that key looks good. And you already created the repo, correct?

**Comment by CharlesSchaack** [comment]
The repository is created, it's at the point where it's showing the quick guide since there is nothing inside the repository yet.

**Comment by CharlesSchaack** [comment]
Do I need to go through the steps of creating the first commit?

**Comment by ericbarch** [comment]
Nope, that should be all that you need to do. Any chance you could test it against GitHub or Bitbucket? Just want to make sure it's not something specific to the Gogs installation.

**Comment by CharlesSchaack** [comment]
Yeah good call, got it working with github. I might have to do some more research into SSH keys. I have key auth setup for my normal SSH on this server and I'm wondering if that is somehow conflicting? I did the same thing for github as I do for gogs, only difference is I removed the -p 2222 and the ssh:// parts of the docker run command.

At least I can get it working with github for now although I'd prefer it to be private so maybe I'll create a bitbucket. I'll just fiddle around and see if I can get it working with gogs over the next few days. Thank you for everything.
